addons/website_slides/models/slide_channel.py

```diff
@@ -216,6 +233,39 @@ def message_post(self, parent_id=False, subtype=None, **kwargs):
                 subtype = 'mail.mt_note'
         return super(Channel, self).message_post(parent_id=parent_id, subtype=subtype, **kwargs)
 
+    # ---------------------------------------------------------
+    # Rating Mixin API
+    # ---------------------------------------------------------
+
+    def action_add_member(self, target_user=False, **member_values):
+        if not target_user:
+            target_user = self.env.user
+        existing = self.env['slide.channel.partner'].sudo().search([
+            ('channel_id', 'in', self.ids),
+            ('partner_id', '=', target_user.partner_id.id)
+        ])
+        to_join = (self - existing.mapped('channel_id'))._filter_add_member(target_user, **member_values)
+        if to_join:
+            self.env['slide.channel.partner'].sudo().create([
+                dict(channel_id=channel.id, partner_id=target_user.partner_id.id, **member_values)
+                for channel in to_join
+            ])
+            return to_join
+        return False
+
+    def _filter_add_member(self, target_user, **member_values):
```

Comment:
Is it necessary to pass target_user and **member_values to this method?

Reply:
(Answering my own comment: I guess it would be easier to have full info if a module is overriding it)

Reply:
Because you may want to create a draft or confirmed member, depending on situation. And currently we are not totally sure about which fields will be available on the decorated m2m (there will be some statistics at least). Having a kwargs eases inheritance even if I agree it is less readable. Maybe once we have a complete working elearning flow we will be able to clean a bit those methods. I will add a fixme.